src/main/java/bio/terra/service/snapshot/flight/create/CreateSnapshotMetadataStep.java

```
@@ -62,7 +62,7 @@ public StepResult doStep(FlightContext context) {
     @Override
     public StepResult undoStep(FlightContext context) {
         logger.debug("Snapshot creation failed. Deleting metadata.");
-        snapshotDao.deleteByName(snapshotReq.getName());
+        snapshotDao.deleteByNameAndFlight(snapshotReq.getName(), context.getFlightId());
```

This is definitely stepping outside of the direct bug you're fixing, but should we be concerned with this delete silently failing? deleteByNameAndFlight returns a success boolean, but we don't check it. Would it be worth adding a log?